src/StatsigOnPrem.ts

```diff
@@ -92,16 +92,16 @@ export default class StatsigOnPrem implements StatsigInterface {
     sdkKey: string,
     options?: ConfigSpecsOptions
   ): Promise<ConfigSpecs> {
-    const registeredKeys = await this.store.getRegisteredSDKKeys();
-    if (!registeredKeys.has(sdkKey)) {
-      console.warn("Attempting to use a non-registered key");
-    }
-
     const cachedSpecs = await this.cache.getSpecs(sdkKey, options);
     if (cachedSpecs) {
       return cachedSpecs;
     }
 
+    const registeredKeys = await this.getRegisteredSDKKeys();
```

Comment:
The main change is using `this.getRegisteredSDKKeys();` instead of `this.store.getRegisteredSDKKeys();` which means the cache is checked first, and will handle if the cache doesnt exist.

This API is already used by `getGlobalSDKKeys`